sbc_email/models/import_letters_history.py

```diff
@@ -379,28 +380,101 @@ def _get_smb_connection(self):
             return SMBConnection(
                 SmbConfig.smb_user, SmbConfig.smb_pass, 'openerp', 'nas')
 
-    @api.model
-    def import_web_letter(self):
+    @api.one
+    def import_web_letter(self, child_code, sponsor_ref, language_name,
+                          original_text, template_name, pdf_letter):
         """
         Call when a letter is set on web site:
             - add web letter to an import set with import letter config
               'Web letter'
         """
-        import_web_letter = self.search([
-            ('config_id.name', '=', 'Web Letter'),
-            ('state', '!=', 'done')])
+        child_id = None
+        if child_code:
+            # Retrieve child code and find corresponding id
+            model_child = self.env['compassion.child'].search(
+                [('code', '=', child_code), ('state', '=', 'P')])
+
+            child_id = model_child.id
+        if not child_id:
+            logger.info('Wrong child_code received : {}'.
+                        format(child_code))
+            return False
+
+        sponsor_id = None
+        if sponsor_ref:
+            # Retrieve sponsor reference and find corresponding id
+            model_sponsor = self.env['res.partner'].search(
+                [('ref', '=', sponsor_ref),
+                 ('is_company', '=', False)])
+
+            sponsor_id = model_sponsor.id
+        if not sponsor_id:
+            logger.info('Wrong sponsor_ref received : {}'.
+                        format(sponsor_ref))
+            return False
+
+        lang_id = None
+        if language_name:
+            # Retrieve language name and find corresponding id
+            model_lang = self.env['res.lang.compassion'].search(
+                [('name', '=', language_name)], limit=1)
+
+            lang_id = model_lang.id
+
+        template = None
+        if template_name:
+            # Retrieve template name and find corresponding id
+            template = self.env['correspondence.template'].search(
+                [('name', '=', template_name)], limit=1)
+
+        if not original_text:
+            original_text = ""
+
+        # save_letter pdf
+        if pdf_letter:
+            file_pdf = BytesIO(base64.b64decode(pdf_letter))
+
+            filename = 'WEB_' + sponsor_ref + '_' + \
+                child_code + '_' + str(time.time())[:10] + '.pdf'
+
+            logger.info("Try to copy file {} !".format(filename))
 
-        if import_web_letter:
-            logger.info('Import letter with config Web Letter existence')
+            # Copy file in attachment in the done letter folder
```

Comment:
I would move the samba code at the end of the function such that we only copy the file at the end after everything went well. In case of error, it doesn't lead to a file stored in done but not imported.